src/Security/RememberLoginHash.php

```diff
@@ -190,14 +202,22 @@ public static function generate(Member $member)
     /**
      * Generates a new hash for this member but keeps the device ID intact
      *
+     * @deprecated 5.3.0 Will be removed without equivalent functionality
      * @return RememberLoginHash
      */
     public function renew()
     {
-        $hash = $this->getNewHash($this->Member());
-        $this->Hash = $hash;
-        $this->extend('onAfterRenewToken');
+        // Only regenerate token if configured to do so
+        Deprecation::notice('5.3.0', 'Will be removed without equivalent functionality');
+        $replaceToken = RememberLoginHash::config()->get('replace_token_during_session_renewal');
+        if ($replaceToken) {
+            $hash = $this->getNewHash($this->Member());
+            $this->Hash = $hash;
+        }
+
+        $this->extend('onAfterRenewToken', $replaceToken);
```

Comment:
@GuySartorelli Just considering how we'll handle this extension point once we drop the method; My current thought is that we shift the call-site up to `CookieAuthenticationHandler::authenticateRequest()`, but still have it be an extension of `RememberLoginHash` (possibly with a tweaked name to reflect that the token itself is not being renewed.)

There's precedent in the `authenticateRequest()` method for calling an extension point externally on an object instance (see the `$member->extend('memberAutoLoggedIn');` call) but not sure if this falls within our general best practices?

Reply:
Isn't the idea with deprecating and eventually removing the `renew()` method that there _will_ be no renewal necessary? If we don't renew the token, we don't need to have a `onAfterRenewToken` extension hook, right?